sycl/test/on-device/online_compiler/online_compiler_common.hpp

```diff
@@ -0,0 +1,181 @@
+#include <CL/sycl.hpp>
+#include <CL/sycl/INTEL/online_compiler.hpp>
+
+#include <iostream>
+#include <vector>
+
+static const char *CLSource = R"===(
+__kernel void my_kernel(__global int *in, __global int *out) {
+  size_t i = get_global_id(0);
+  out[i] = in[i]*2 + 100;
+}
+)===";
+
+static const char *CLSourceSyntaxError = R"===(
+__kernel void my_kernel(__global int *in, __global int *out) {
+  syntax error here
+  size_t i = get_global_id(0);
+  out[i] = in[i]*2 + 100;
+}
+)===";
+
+static const char *CMSource = R"===(
+extern "C"
+void cm_kernel() {
+}
+)===";
+
+using namespace sycl::INTEL;
+
+#ifdef RUN_KERNELS
+void testSyclKernel(sycl::queue &Q, sycl::kernel Kernel) {
+  std::cout << "Run the kernel now:\n";
+  const int N = 4;
+  int InputArray[N] = {0, 1, 2, 3};
+  int OutputArray[N] = {};
+
+  sycl::buffer<int, 1> InputBuf(InputArray, sycl::range<1>(N));
```

Comment:
I have the feeling that there is an Intel extension for
```suggestion
  sycl::buffer InputBuf { InputArray};
```
which is now in SYCL 2020